### PR TITLE
[IOTDB-221] Add C++ client

### PR DESCRIPTION
**Pull request by humanfy**

### CODE REVIEW

File: client-cpp/src/IOTDBSession.cpp [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    TSCloseSessionReq* req = new TSCloseSessionReq();
+    req->__set_sessionHandle(*sessionHandle);
+    try 
+    {
+        TSStatus *resp = new TSStatus();
+        client->closeSession(*resp,*req);
+    } 
+    catch (TException e) 
+    {
+        throw IoTDBSessionException("Error occurs when closing session at server. Maybe server is down.");
```
Comment (jt2594838) [inline client-cpp/src/IOTDBSession.cpp]: Better to preserve the error message in e.

File: client-cpp/compile.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
```
Comment (chrisdutz) [inline client-cpp/compile.sh]: This would result in a build that excludes Windows users.

File: client-cpp/src/SessionExample.cpp [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    
+    vector<string> paths;
+    paths.push_back("root.sg1.d1.s1");
+    paths.push_back("root.sg1.d1.s2");
+    paths.push_back("root.sg1.d1.s3");
+    session->deleteTimeseries(paths);
+    
+    session->deleteStorageGroup("root.sg1");
+    
+    session->close();
```
Comment (jt2594838) [inline client-cpp/src/SessionExample.cpp]: Please show how to perform a query, which is important.

**Comment by jixuan1989** [comment]
Hi, I will close this PR because a new c++ module is added to the master branch in #1616

**Pull request closed without merging**